packages/cli/lib/validators/logValidatorResults.js

```diff
@@ -0,0 +1,51 @@
+const { logger } = require('@hubspot/cli-lib/logger');
+const { VALIDATION_RESULT } = require('./constants');
+
+function logResultsAsJson(results) {
+  let success = true;
+  const resultObj = results.reduce((acc, result) => {
```

Comment:
If I'm following correctly, it seems like we're taking the `results` array, applying a bit of logic to each element, and returning another array as the accumulator. Just curious if there's a reason to use `reduce` rather than `map`

Reply:
Yeah, I'm using reduce here because I'm actually building a nested object that has a slightly different shape than the input list. I might be able to use map, but reduce felt more natural. It looks kind of like this:
```js
// Input
[
  { validatorKey: 'themeConfig', result: 'FATAL', error: 'blah1' },
  { validatorKey: 'themeConfig', result: 'WARNING', error: 'blah2' },
  { validatorKey: 'templateLimit', result: 'FATAL', error: 'blah3' },
]

// Output
{
  themeConfig: [ { result: 'FATAL', error: 'blah1' },  { result: 'WARNING', error: 'blah2' } ],
  templateLimit: [ { result: 'FATAL', error: 'blah3' } ],
}
```

Reply:
Oh gotcha, we're consolidating all of the results based on their respective validation types